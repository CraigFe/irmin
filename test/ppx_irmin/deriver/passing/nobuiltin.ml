(* When a type is annotated with the [nobuiltin] annotation, it should be
   considered as an abstract type (i.e. don't pull representations from
   [Irmin.Type]). *)

type unit = string [@@deriving irmin]

type t = (unit[@nobuiltin]) [@@deriving irmin]

type foo = (unit[@irmin.nobuiltin]) [@@deriving irmin]

(* TODO: uncomment once https://github.com/mirage/irmin/pull/970 is merged *)

(* let (_ : unit_t Irmin.Type.t) = t
 * 
 * let (_ : unit_t Irmin.Type.t) = foo_t *)
