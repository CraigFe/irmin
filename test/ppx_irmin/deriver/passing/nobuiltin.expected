type unit = string[@@deriving irmin]
include struct let unit_t = Irmin.Type.string end[@@ocaml.doc "@inline"]
[@@merlin.hide ]
type t = ((unit)[@nobuiltin ])[@@deriving irmin]
include struct let t = unit_t end[@@ocaml.doc "@inline"][@@merlin.hide ]
type foo = ((unit)[@irmin.nobuiltin ])[@@deriving irmin]
include struct let foo_t = Irmin.Type.unit end[@@ocaml.doc "@inline"]
[@@merlin.hide ]
let (_ : unit_t Irmin.Type.t) = t
let (_ : unit_t Irmin.Type.t) = foo_t
